src/components/molecules/NavBar/NavBar.tsx

```diff
@@ -91,20 +91,25 @@ const NavBar: React.FC<NavBarPropsType> = ({ redirectionUrl }) => {
   const { user, profile } = useUser();
   const venueId = useVenueId();
   const venue = useSelector(currentVenueSelectorData);
+  const venueParentId = venue?.parentId;
   const privateChats = useSelector(privateChatsSelector);
   const radioStations = useSelector(radioStationsSelector);
   const parentVenue = useSelector(parentVenueSelector);
 
-  const venueParentId = venue?.parentId;
-  const venueParentQuery = useMemo<ReduxFirestoreQuerySetting>(
-    () => ({
-      collection: "venues",
-      doc: venueParentId,
-      storeAs: "parentVenue",
-    }),
-    [venueParentId]
+  const venueParentQuery = useMemo<SparkleRFQConfig[]>(
+    () =>
+      venue?.parentId
+        ? [
+            {
+              collection: "venues",
+              doc: venue.parentId,
+              storeAs: "parentVenue",
+            },
+          ]
+        : [],
+    [venue]
```

Comment:
We don't need to `useMemo` on these queries. This was an earlier assumption we made that turned out to be wrong.

Since the reason it was split out of the `useSparkleFirestoreConnect` (memoisation) doesn't even matter anymore, so if it's not used elsewhere, we can just combine it together again (see suggest change in other comment below)

```suggestion
```